tools/storm/scripts/acr.go

```diff
@@ -0,0 +1,186 @@
+package scripts
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type AcrScriptSet struct {
+	// This field represents the "acr-push" subcommand.
+	AcrPush AcrPushScript `cmd:"" help:"Pushes images to an ACR"`
+
+	// This field represents the "acr-delete" subcommand.
+	AcrDelete AcrDeleteScript `cmd:"" help:"Deletes images from an ACR"`
+}
+
+func generateTagBase(buildId string, config string, deploymentEnvironment string) string {
+	return fmt.Sprintf("v%s.%s.%s", buildId, config, deploymentEnvironment)
+}
+
+func loginToACR(acrName string) error {
+	logrus.Infof("Logging in to ACR: %s", acrName)
+	cmd := exec.Command("az", "acr", "login", "-n", acrName)
+	return cmd.Run()
+}
+
+// Define AcrPushScript
+type AcrPushScript struct {
+	Config                string   `required:"" help:"Trident configuration (e.g., 'extensions')" type:"string"`
```

Comment:
re: `type:"string"`
golang is already a strongly and statically typed language, you are telling the parser that this is a string with the field's type definition.

in the context of kong, `type` refers to type mappers, you can think of these as "specializations" of the type. Valid values are here: https://github.com/alecthomas/kong?tab=readme-ov-file#custom-named-decoders

For this I _think_ you want the `existingfile` type. An extra advantage of using this type is that it checks that the file exists for you, no need to re-check manually. :)